### PR TITLE
Added CLI tests for colored output

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

File: test/cli/color-tty.test.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                done(code);
+            });
+        });
+    });
+
+    describe('noTTY', function () {
+        var encoding = 'utf-8';
+
+        before(function () {
+            try { fs.mkdirSync('out'); }
```
Comment (shamasis) [inline test/cli/color-tty.test.js]: `before` can use `done` callback. Use async mkdir.

File: test/cli/color-tty.test.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            catch (e) { console.error(e); }
+        });
+
+        it('should not produce colored output without any options', function (done) {
+            exec('node ./bin/newman.js run test/cli/single-get-request.json > out/notty.txt',
+            function (code, stdout, stderr) {
+                expect(code).be(0);
+                expect(stderr).to.be.empty();
+                expect(stdout).to.be.empty();
+                expect(fs.readFileSync('out/notty.txt', encoding)).to.not.match(coloredOutput);
```
Comment (shamasis) [inline test/cli/color-tty.test.js]: please use async readFile.

```
fs.readFile('out/notty.txt', function (err, buf) {
    if (err) { return done(err); }
    expect()...
});

File: test/cli/color-tty.test.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            });
+        });
+
+        it('should produce colored output with --color', function (done) {
+            exec('node ./bin/newman.js run test/cli/single-get-request.json --color > out/notty-color.txt',
+            function (code, stdout, stderr) {
+                expect(code).be(0);
+                expect(stderr).to.be.empty();
+                expect(stdout).to.be.empty();
+                expect(fs.readFileSync('out/notty-color.txt', encoding)).to.match(coloredOutput);
```
Comment (shamasis) [inline test/cli/color-tty.test.js]: async fs please

File: test/cli/color-tty.test.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            });
+        });
+
+        it('should not produce colored output with --no-color', function (done) {
+            exec('node ./bin/newman.js run test/cli/single-get-request.json --no-color > out/notty-no-color.txt',
+            function (code, stdout, stderr) {
+                expect(code).be(0);
+                expect(stderr).to.be.empty();
+                expect(stdout).to.be.empty();
+                expect(fs.readFileSync('out/notty-no-color.txt', encoding)).to.not.match(coloredOutput);
```
Comment (shamasis) [inline test/cli/color-tty.test.js]: async fs please

File: test/cli/color-tty.test.js (hunk 1 above)
Comment (kunagpal) [inline test/cli/color-tty.test.js]: Done.

File: test/cli/color-tty.test.js (hunk 2 above)
Comment (kunagpal) [inline test/cli/color-tty.test.js]: Done.

File: test/cli/color-tty.test.js (hunk 3 above)
Comment (kunagpal) [inline test/cli/color-tty.test.js]: Done.

File: test/cli/color-tty.test.js (hunk 4 above)
Comment (kunagpal) [inline test/cli/color-tty.test.js]: Done.

File: test/cli/color-tty.test.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    });
+
+    describe('noTTY', function () {
+        var encoding = 'utf-8',
+            outFile = 'out/color-tty-test.txt';
+
+        beforeEach(function (done) {
+            fs.stat('out', function (err) {
+                if (err) {
+                    console.error(err);
```
Comment (shamasis) [inline test/cli/color-tty.test.js]: remove this line

File: test/cli/color-tty.test.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+                    expect(data).to.not.match(coloredOutput);
+                    done();
+                });
+            });
+        });
+
+        afterEach(function (done) {
+            fs.stat(outFile, function (err) {
+                if (err) {
+                    console.error(err);
```
Comment (shamasis) [inline test/cli/color-tty.test.js]: remove line

**Pull request merged**